controllers/topics/inviteMembers.js

```diff
@@ -90,23 +90,25 @@ module.exports = async (req, res) => {
           }
         });
 
-        const payload = {
-          notification: {
-            title: `#${topics?.name}`,
-            body: invitations_msg,
-            image: user.profile_pic_path
-          },
-          data: {
-            topic_id,
-            type: 'topic',
-            topic_name: topics?.name
-          }
-        };
         if (userTokens) {
           // looping for each user token
           userTokens.forEach((userToken) => {
+            const payload = {
+              notification: {
+                title: `#${topics?.name}`,
+                body: invitations_msg,
+                image: user.profile_pic_path
+              },
+              data: {
+                topic_id,
+                type: 'topic',
+                topic_name: topics?.name
+              },
+              token: userToken?.token
+            };
+
             messaging()
-              .sendToDevice(userToken.token, payload)
+              .send(payload)
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new code looks good overall, but there's a potential issue with error handling. The promise returned by `messaging().send(payload)` is not properly handled. If the `send` function fails for any reason (e.g., network issues, invalid token), the error will be unhandled and could crash your application. It would be better to add a `.catch()` block to handle any errors that might occur.
```diff
             messaging()
              .send(payload)
-               .then(() => {});
+               .then(() => {})
+               .catch((error) => {
+                 console.error(`Failed to send notification: ${error.message}`);
+               });
```
This way, you can log the error message and prevent it from crashing your application.


<!-- This is an auto-generated comment by OSS CodeRabbit -->